src/main/java/org/seqdoop/hadoop_bam/BAMSplitGuesser.java

```diff
@@ -186,7 +187,8 @@ public long guessNextBAMRecordStart(long beg, long end)
 						if (record == null) {
 							break;
 						}
-						record.getCigar(); // force decoding of CIGAR
+						record.setHeader(header);
```

Comment:
One suggestion to make this more robust would be to use `setHeaderStrict` instead of `setHeader`. `setHeader` is tolerant of a record with a reference index or mate reference index that can't be resolved against the header's sequence dictionary. `setHeaderStrict` is identical, but throws `IllegalArgumentException` if either one can't be resolved against the header. So in that sense its a "safe" way to quickly reject a bad split guess before calling eagerDecode.

Reply:
@cmnbroad good idea! I've made this change.